---
title: "DIFFERENTIAL EXPRESSION ANALYSIS"
author: "Laura"
date: "2023-01-30"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
# Install and Load required packages
```{r Install Packages, include=FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("DESeq2", force = TRUE)
BiocManager::install("apeglm")
BiocManager::install("biomaRt")
BiocManager::install("biomartr")
install.packages("tidyverse")
install.packages("pheatmap")
install.packages(c("gplots", "amap"))
devtools::install_github("kevinblighe/EnhancedVolcano")
BiocManager::install("apeglm")
BiocManager::install("clusterProfiler")
BiocManager::install("org.At.tair.db")
install.packages("DT")
```
```{r Load Packages, include=FALSE}
library("DESeq2")
library("apeglm")
library("pheatmap")
library("tidyverse")
require("vidger")
library("biomaRt")
library("ggplot2")
library("EnhancedVolcano")
library("RColorBrewer")
library("gplots")
library("amap")
library("BiocParallel")
library("ggpubr")
library("easypackages")
library("readr")
library("viridis")
library("clusterProfiler")
library("enrichplot")
library("DT")
suppressPackageStartupMessages(library("org.At.tair.db"))
```
# create an object with the directory containing HTseq counts:
```{r}
# Create an object with the directory containing HTseq counts:
directory <- "C:/Users/laura/Desktop/Master/gdav/DEA"
#list.files(directory)
sampleFiles <- list.files(directory, pattern = ".*\\.count")
sampleFiles

# create a vector of sample names. 
sampleNames <- c( "hightemp01", "hightemp02","normal01","normal02")

# create a vector of conditions. 
replicate <- c("Rep1","Rep2","Rep1","Rep2")

# create a vector of conditions. 
sampleCondition <- c("hightemp","hightemp","normal", "normal")

# now create a data frame from these three vectors. 
sampleTable <- data.frame(
		sampleName = sampleNames,
		fileName = sampleFiles,
		condition = sampleCondition,
		replicate = replicate)

sampleTable
```

```{r}
## Make DESeq2 object from counts and metadata
ddsHTSeq <- DESeqDataSetFromHTSeqCount(
		sampleTable = sampleTable, 
		directory = directory, 
		design = ~condition)
ddsHTSeq
ddsHTSeq$condition

#specifying the reference level:
ddsHTSeq$condition <- relevel(ddsHTSeq$condition, ref = "normal")
ddsHTSeq$condition
```

```{r}
#Perform a minimal pre-filtering to keep only rows that have at least 10 reads total.
# sum counts for each gene across samples
sumcounts <- rowSums(counts(ddsHTSeq))

# take the log
logsumcounts <- log(sumcounts,base=10)

# plot a histogram of the log scaled counts
plot_without_filter <- hist(logsumcounts,breaks=100, main="Histogram of the log scaled counts")

# get genes with summed counts greater than 10; remove lowly expressed genes
keep <- sumcounts > 10

# keep only the genes for which the vector "keep" is TRUE
ddsHTSeq_filter <- ddsHTSeq[keep,]
sumcounts <- rowSums(counts(ddsHTSeq_filter))

# take the log
logsumcounts <- log(sumcounts,base=10)

# plot a histogram of the log scaled counts
plot_with_filter <- hist(logsumcounts,breaks=100,main="Histogram of the log scaled counts \n keep only the genes with summed counts greater than 10" )
```

```{r}
dds <- DESeq(ddsHTSeq_filter)
plotDispEsts(dds, main="Dispersion plot")

# get results table
# get basic statistics about the number of significant genes
# treated vs control
res <- results(dds, pAdjustMethod="BH") 
res_DF <- as.data.frame(res)
resultsNames(dds)
summary(res)

# check out the first few lines
head(res)
mcols(res, use.names = T)
resultsNames(dds)
```

```{r}
##Create normalized read counts
normalized_counts <- counts(dds, normalized=TRUE)
head(normalized_counts)
normalized_counts_mad <- apply(normalized_counts, 1, mad)
normalized_counts <- normalized_counts[order(normalized_counts_mad, decreasing=T), ]
as.data.frame(normalized_counts)

#DESeq get results table
Res_A_X_total <- results(dds, name="condition_hightemp_vs_normal", pAdjustMethod="BH")
Res_A_X_total <- Res_A_X_total[order(Res_A_X_total$padj),]
Res_A_X_total <- data.frame(Res_A_X_total)
Res_A_X_total  <- rownames_to_column(Res_A_X_total, var = "ensembl_gene_id")
nrow(Res_A_X_total)

#rownames(Res_A_X_total)
#Export output files
Res_A_X_total$sig <- ifelse(Res_A_X_total$padj < 0.05, "yes", "no")
Res_A_X_tota_0.05 <-subset(Res_A_X_total, padj < 0.05)
write.csv(normalized_counts,"Condition_hightemp_vs_WT_normcounts.csv")
write.csv(Res_A_X_total, "Condition_hightemp_vs_WT_results_total.csv")
write.csv(Res_A_X_tota_0.05, "Condition_hightemp_vs_WT_result_padj_0.05.csv") 

Res_A_X_total$sig <- ifelse(Res_A_X_total$padj < 0.01, "yes", "no")
Res_A_X_tota_0.01 <-subset(Res_A_X_total, padj < 0.01)
```

## RNA-seq data quality assessment

### Sample Clustering {.tabset .tabset-fade .tabset-pills}

Hierarchical clustering RNA-seq gene expression patterns based on sample distance (Euclidean distance matrix) of log transformed expression from transcripts based on DeSeq2 (Love et al. 2014). 

```{r, warning = FALSE, message = FALSE, echo = FALSE}
# log transformed expression

#This function transforms the count data to the log2 scale in a way which minimizes differences between samples for rows with small counts, and which normalizes with respect to library size. The rlog transformation produces a similar variance stabilizing effect as varianceStabilizingTransformation, though rlog is more robust in the case when the size factors vary widely. The transformation is useful when checking for outliers or as input for machine learning techniques such as clustering or linear discriminant analysis. rlog takes as input a DESeqDataSet and returns a RangedSummarizedExperiment object.

rld <- rlog(dds, blind=FALSE)
rlogMat <- assay(rld)
rlogMat <- rlogMat[order(normalized_counts_mad, decreasing=T), ]

# Generar color
hmcol <- colorRampPalette(brewer.pal(9, "GnBu"))(100)

 # Pearson correlation
pearson_cor <- as.matrix(cor(rlogMat, method="pearson"))

 # Hierarchical clustering
hc <- hcluster(t(rlogMat), method="pearson")

TreeC = as.dendrogram(hc, method="average")
plot(TreeC,
     main = "Sample Clustering-checking outliers ",
     ylab = "Height")

heatmap.2(pearson_cor, Rowv=as.dendrogram(hc), symm=T, trace="none",
col=hmcol, margins=c(11,11), main="Pearson correlation of each
sample")

# normalized, variance-stabilized transformed counts for visualization
vsd <- vst(dds, blind=FALSE)

plotPCA(vsd, intgroup="condition") + geom_text(aes(label=name),vjust=-1) +  ggtitle("PCA Plot using variance stabilizing transformation data") + scale_x_continuous(limits = c(-20, 20))  +  scale_y_continuous(limits = c(-11, 11))  # change limits to fix figure dimensions

plotPCA(rld, intgroup="condition") + geom_text(aes(label=name),vjust=1.5) + ggtitle("PCA Plot using Regularized log  Transformation data")+ scale_x_continuous(limits = c(-20, 20))  +  scale_y_continuous(limits = c(-10, 10))  # change limits to fix figure dimensions


pca_data <- plotPCA(rld, intgroup="condition", returnData=T)

```

### PlotMA {.tabset .tabset-fade .tabset-pills}

In DESeq2, the function plotMA shows the log2 fold changes attributable to a given variable over the mean of normalized counts for all the samples in the DESeqDataSet. Points will be colored red if the adjusted p value is less than 0.05. Points which fall out of the window are plotted as open triangles pointing either up or down.


```{r, warning = FALSE, message = FALSE, echo = FALSE, fig.width=5, fig.height=5}

plotMA(dds , alpha = 0.05, main=("Standard Bland-Altman- MA Plot"))


#Shrinkage of effect size (LFC estimates) is useful for visualization and ranking of genes. It is more useful visualize the MA-plot for the shrunken log2 fold changes, which remove the noise associated with log2 fold changes from low count genes without requiring arbitrary filtering thresholds. This helps to get more meaningful log2 fold changes for all genes independently of their expression level.

resLFC <- lfcShrink(dds, coef="condition_hightemp_vs_normal", type="apeglm")

#You can see that genes with low counts are now shrinked.                  
plotMA(resLFC, alpha = 0.05, ylim=c(-4,4) ,main= ("MA plot shrunken log2 fold changes"))


maplot = ggmaplot(resLFC, size = 1,
                  palette = c("#e55c30", "#84206b", "#f6d746"),
                  genenames = as.vector(rownames(resLFC)),
                  legend="top", top = 15,font.label = c("bold", 11),
                  font.legend = c("bold",12), font.main = "bold",
                  xlab = "Log2 Mean Expression",  ylab="Log2 FC")
show(maplot)

```

### Volcano plot {.tabset .tabset-fade .tabset-pills}

```{r, warning = FALSE, message = FALSE, echo = FALSE}
#For each gene, this plot shows the gene fold change on the x-axis against the p-value plotted on the y-axis.

#Here, we make use of a library called EnhancedVolcano which is available through Bioconductor and described extensively on its own GitHub page.

# The main function is named after the package
EnhancedVolcano(toptable = resLFC,      # We use the shrunken log2 fold change as noise associated with low count genes is removed 
                x = "log2FoldChange",   # Name of the column in resLFC that contains the log2 fold changes
                y = "padj",             # Name of the column in resLFC that contains the p-value
                pCutoff = 0.05,
                FCcutoff = 1.3,
                xlim = c(-2, 4),
                title="Condition hightemp vs. normaltemp", subtitle = NULL,
                lab = rownames(resLFC))


# Gather Log-fold change and FDR-corrected pvalues from DESeq2 results
## - Change pvalues to -log10 (1.3 = 0.05)
data <- data.frame(gene = row.names(res),
                   pval = -log10(res$padj), 
                   lfc = res$log2FoldChange)

# Remove any rows that have NA as an entry
data <- na.omit(data)

# Color the points which are up or down
## If fold-change > 0 and pvalue > 1.3 (Increased significant)
## If fold-change < 0 and pvalue > 1.3 (Decreased significant)
data <- mutate(data, color = case_when(data$lfc > 0 & data$pval > 1.3 ~ "Increased",
                                       data$lfc < 0 & data$pval > 1.3 ~ "Decreased",
                                       data$pval < 1.3 ~ "nonsignificant"))

# Make a basic ggplot2 object with x-y values
vol <- ggplot(data, aes(x = lfc, y = pval, color = color))

# Add ggplot2 layers
vol +   
  ggtitle(label = "Volcano Plot", subtitle = "Colored by fold-change direction") +
  geom_point(size = 2.5, alpha = 0.8, na.rm = T) +
  scale_color_manual(name = "Directionality",
                     values = c(Increased = "#008B00", Decreased = "#CD4F39", nonsignificant = "darkgray")) +
  theme_bw(base_size = 14) + # change overall theme
  theme(legend.position = "right") + # change the legend
  xlab(expression(log[2]("LoGlu" / "HiGlu"))) + # Change X-Axis label
  ylab(expression(-log[10]("adjusted p-value"))) + # Change Y-Axis label
  geom_hline(yintercept = 1.3, colour = "darkgrey") + # Add p-adj value cutoff line
  scale_y_continuous(trans = "log1p") # Scale yaxis due to large p-values
```

### Pheatmap {.tabset .tabset-fade .tabset-pills}
```{r}
signi <- subset (Res_A_X_total,padj <=0.05)

normcounts_A_X <- data.frame(normalized_counts)
normcounts_A_X <- rownames_to_column(normcounts_A_X, var = "ensembl_gene_id")
allsig <- merge(normcounts_A_X, signi,by='ensembl_gene_id')

sigCounts <- allsig [, 2:7] #CUIDADO!!

row.names(sigCounts) <- allsig$Row.names

pheatmap (log2(sigCounts+ 1), scale = "row", show_rownames = F, treeheight_row = 0, treeheight_col =0, main = ("Clustering using significant genes") )

# get the top 20 genes by shrunken log2 fold change
top20 <- order(-abs(resLFC$log2FoldChange))[1:20]
#resLFC[top20,]


# negative log-scaled adjusted p-values
log_padj <- -log(resLFC$padj,10)
log_padj[log_padj > 100] <- 100


# get top 20 log fold change genes
top50 <- order(-abs(resLFC$log2FoldChange))[1:20]
df <- data.frame(colData(dds)[,"condition"])
	rownames(df) <- colnames(dds)
	colnames(df) <- "condition"
pheatmap(
	assay(rld)[top50,], 
	cluster_rows=TRUE, 
	show_rownames=TRUE,
	cluster_cols=FALSE,
	annotation_col=df,
	main= ("Clustering using top 20 log fold change genes")
	)
```

```{r}
Alteredgenes <- Res_A_X_tota_0.01$ensembl_gene_id
Description <- c("Nif-specific regulatory protein","FeMo cofactor biosynthesis protein NifB", "Nitrogenase iron protein 1", "hypothetical protein", "hypothetical protein")
Name <- c("nifA", "nifB", "nifH1", "-", "-")
pval <- Res_A_X_tota_0.01$pvalue
padj <- Res_A_X_tota_0.01$padj
foldchange <- Res_A_X_tota_0.01$log2FoldChange
AnnotationTable <- data.frame(Alteredgenes, Description, Name, pval, padj, foldchange)
AnnotationTable
```











